client/src/components/timetable/AddCourse.jsx

```diff
@@ -34,58 +34,51 @@ class AddCourse extends Component {
     this.setState({ current: updatedlist });
   }
 
-  handleCourseAddition(input) {
-    let selectedCode = input.target.id;
-    let selectCourse = obj =>
-      Object.keys(obj)
-        .filter(course => course === selectedCode)
-        .reduce((res, key) => ((res[key] = obj[key]), res), {});
-    let selectedCourse = selectCourse(this.state.current);
-    this.props.updateCurrent(selectedCourse);
-    this.setState({
-      selectedCourse: selectedCourse
-    });
-  }
-
   getSections(type) {
-    let course = this.state.selectedCourse;
+    let course = this.props.currentCourse;
     let code = Object.keys(course)[0];
     let selectSections = obj =>
       Object.keys(obj)
         .filter(item => item.charAt(0) === type)
         .reduce((res, key) => ((res[key] = obj[key]), res), {});
-    let list = selectSections(this.state.selectedCourse[code].sections);
-    return list;
+    return selectSections(this.props.currentCourse[code].sections);
   }
 
-  changeCourse() {
-    this.setState({ selectedCourse: null, current: this.props.allCourses });
+  clearCourse() {
+    this.props.clearCurrentCourse();
+    this.setState({ current: this.state.initial });
   }
 
   render() {
     return (
       <div>
-        {!this.state.selectedCourse ? (
+        {!this.props.currentCourse ? (
           <div>
             <Search action={this.filterItems} />
-            <ListCourse
-              courses={this.state.current}
-              action={this.handleCourseAddition}
-            />
+            <ListCourse courses={this.state.current} />
           </div>
         ) : (
           <div>
-            <h3>{Object.keys(this.state.selectedCourse)}</h3>
-            <ToggleButton action={this.changeCourse} title='Change Course' />
-            <SectionTabs
-              action={this.props.addSection}
-              getSections={this.getSections}
-            />
+            <h3>{Object.keys(this.props.currentCourse)}</h3>
+            <ToggleButton action={this.clearCourse} title="Change Course" />
+            <SectionTabs getSections={this.getSections} />
           </div>
         )}
       </div>
     );
   }
 }
 
-export default AddCourse;
+const mapStateToProps = state => {
+  return {
+    currentCourse: state.updateCC.currentCourse
+  };
+};
+
+const mapDispatchToProps = dispatch => {
+  return {
+    clearCurrentCourse: () => dispatch(clearCurrentCourse())
+  };
+};
+
```

Comment:
Maybe add a bit of proptype validation here, just so no one uses this component incorrectly by mistake?

Reply:
imo its fine enough!

Reply:
It's fine, it will work. Awesome.

Only thing is it makes the code safer. In fact, this should be done for all components with props.

Reply:
yeah! I took prototype validation in wrong sense.Will add propTypes later